src/main/java/com/openlattice/chronicle/services/enrollment/EnrollmentManagerImpl.java

```diff
@@ -0,0 +1,389 @@
+    }
+
+    @Override
+    public boolean isKnownDatasource(
+            UUID organizationId, UUID studyId, String participantId, String datasourceId ) {
+
+        return getDeviceEntityKeyId( organizationId, studyId, participantId, datasourceId ) != null;
+    }
+
+    @Override
+    public boolean isKnownParticipant( UUID organizationId, UUID studyId, String participantId ) {
+        return commonTasksManager.getParticipantEntityKeyId( organizationId, studyId, participantId ) != null;
+    }
+
+    @Override
+    public Map<FullQualifiedName, Set<Object>> getParticipantEntity(
+            UUID organizationId, UUID studyId, UUID participantEntityKeyId ) {
+        try {
+            ApiClient apiClient = apiCacheManager.prodApiClientCache.get( ApiClient.class );
+            DataApi dataApi = apiClient.getDataApi();
+
+            UUID entitySetId = commonTasksManager.getParticipantEntitySetId( organizationId, studyId );
+            if ( entitySetId == null ) {
+                logger.error( "Unable to load participant EntitySet id." );
+                return ImmutableMap.of();
+            }
+            Map<FullQualifiedName, Set<Object>> entity = dataApi.getEntity( entitySetId, participantEntityKeyId );
+            if ( entity == null ) {
+                logger.error( "Unable to get participant entity." );
+                return ImmutableMap.of();
+            }
+            return entity;
+
+        } catch ( ExecutionException e ) {
+            logger.error( "Unable to get participant entity.", e );
+            return ImmutableMap.of();
+        }
+    }
+
+    @Override
+    public ParticipationStatus getParticipationStatus(
+            UUID organizationId, UUID studyId, String participantId ) {
+        logger.info( "getting participation status: orgId = {}, studyId = {}, participantId = {}",
+                organizationId,
+                studyId,
+                participantId );
+
+        try {
+            ApiClient apiClient = apiCacheManager.prodApiClientCache.get( ApiClient.class );
+            SearchApi searchApi = apiClient.getSearchApi();
+
+            // entity set ids
+            UUID studiesESID = commonTasksManager.getEntitySetId( organizationId, CHRONICLE, STUDIES, STUDY_ES );
+            UUID participatedInESID = commonTasksManager.getEntitySetId( organizationId, CHRONICLE, PARTICIPATED_IN,
+                    PARTICIPATED_IN_ES );
+            UUID participantsESID = commonTasksManager.getParticipantEntitySetId( organizationId, studyId );
+
+            checkNotNullUUIDs( Sets.newHashSet( studiesESID, participatedInESID, participantsESID ) );
+
+            // participant must exist
+            UUID participantEKID = Preconditions
+                    .checkNotNull( commonTasksManager
+                                    .getParticipantEntityKeyId( organizationId, studyId, participantId ),
+                            "participant not found" );
```

Comment:
probably helpful to log participantId in the error here (plus maybe org and study ids)